pkg/controllers/cert/cert_rotating_controller.go

```diff
@@ -0,0 +1,287 @@
+func (c *CertRotatingController) Reconcile(ctx context.Context, req controllerruntime.Request) (controllerruntime.Result, error) {
+	klog.V(4).Infof("Rotating the karmada-agent certificate of cluster: %s", req.NamespacedName.Name)
+
+	cluster := &clusterv1alpha1.Cluster{}
+	if err := c.Client.Get(context.TODO(), req.NamespacedName, cluster); err != nil {
+		// The resource may no longer exist, in which case we stop processing.
+		if apierrors.IsNotFound(err) {
+			return controllerruntime.Result{}, nil
+		}
+
+		return controllerruntime.Result{Requeue: true}, err
+	}
+
+	if !cluster.DeletionTimestamp.IsZero() {
+		return controllerruntime.Result{}, nil
+	}
+
+	return c.syncClusterCert(cluster)
+}
+
+// SetupWithManager creates a controller and register to controller manager.
+func (c *CertRotatingController) SetupWithManager(mgr controllerruntime.Manager) error {
+	return controllerruntime.NewControllerManagedBy(mgr).
+		For(&clusterv1alpha1.Cluster{}, builder.WithPredicates(c.PredicateFunc)).
+		WithEventFilter(predicate.GenerationChangedPredicate{}).
+		WithOptions(controller.Options{
+			RateLimiter: ratelimiterflag.DefaultControllerRateLimiter(c.RatelimiterOptions),
+		}).
+		Complete(c)
+}
+
+func (c *CertRotatingController) syncClusterCert(cluster *clusterv1alpha1.Cluster) (controllerruntime.Result, error) {
+	var err error
+	// create a ClusterClient for the given member cluster
+	c.ClusterClient, err = c.ClusterClientSetFunc(cluster.Name, c.Client, c.ClusterClientOption)
+	if err != nil {
+		klog.Errorf("Failed to create a ClusterClient for the given member cluster: %v, err is : %v", cluster.Name, err)
+		return controllerruntime.Result{}, fmt.Errorf("failed to create a ClusterClient: %w", err)
+	}
+
+	secret, err := c.ClusterClient.KubeClient.CoreV1().Secrets("karmada-system").Get(context.TODO(), "karmada-kubeconfig", metav1.GetOptions{})
+	if err != nil {
+		return controllerruntime.Result{}, fmt.Errorf("failed to get karmada kubeconfig: %w", err)
+	}
+
+	err = c.syncCertRotating(secret)
+	if err != nil {
+		return controllerruntime.Result{Requeue: true}, err
+	}
+	// msg := fmt.Sprintf("Sync work (%s) to cluster(%s) successful.", work.Name, clusterName)
+	// klog.V(4).Infof(msg)
+	// c.EventRecorder.Event(work, corev1.EventTypeNormal, workv1alpha1.EventReasonSyncWorkSucceed, msg)
+	return controllerruntime.Result{}, nil
+}
+
+func (c *CertRotatingController) syncCertRotating(secret *corev1.Secret) error {
+	if secret.Data == nil {
+		return fmt.Errorf("no client certificate found in secret %q", secret.Namespace+"/"+secret.Name)
+	}
+
+	karmadaKubeconfigData, ok := secret.Data["karmada-kubeconfig"]
+	if !ok {
+		return fmt.Errorf("no karmada kubeconfig found in secret %q", secret.Namespace+"/"+secret.Name)
+	}
+
+	config, err := clientcmd.Load(karmadaKubeconfigData)
+	if err != nil {
+		return err
+	}
+
+	clusterName := config.Contexts[config.CurrentContext].AuthInfo
```

Comment:
For security, should we check whether it exists this key `config.CurrentContext`.